cpp/ql/test/README.md

```diff
@@ -0,0 +1,53 @@
+# C/C++ CodeQL tests
+
+This document provides additional information about the C/C++ CodeQL Tests located in `cpp/ql/test`.  See [Contributing to CodeQL](CONTRIBUTING.md) for general information about contributing to this repository.
+
+The tests can be run through Visual Studio Code.  Advanced users may also use the `codeql test run` command.
+
+## Contributing to the tests
+
+We are keen to have unit tests for all of our QL code.
+
+Every query in `cpp/ql/src` (outside of `cpp/ql/src/experimental`) should have a test in the corresponding subdirectory of `cpp/ql/test/query-tests`. At a minimum, each query test shall contain one case that should be detected by the query, and one related case that should not.
+
+For example a simple test for the "Memory is never freed" (`cpp/memory-never-freed`) query might contain the following cases:
+```
+int *array1, *array2;
+
+array1 = (int *)malloc(sizeof(int) * 100); // BAD: never freed
+
+array2 = (int *)malloc(sizeof(int) * 100); // GOOD
+free(array2);
+```
+
+Features of the QL libraries in `cpp/ql/src` should also have test coverage, in `cpp/ql/test/library-tests`.
+
+## Copying code
+
+The contents of `cpp/ql/test` should be original - nothing should be copied from other sources. In particular do not copy-paste C/C++ code from third party projects, your own projects, or the standard C/C++ library implementation of your compiler (regardless of the associated license). As an exception, required declarations may be taken from the following sources where necessary:
+ - ISO/IEC Programming languages - C (all versions)
+ - ISO/IEC Programming languages - C++ (all versions)
```

Comment:
I think I prefer the links to specific versions, even though they will become outdated, because as a user I'd prefer to be linked to something specific over a link to a search.

There are also draft versions, which I believe are free and thus probably quite popular in practice (despite potentially being inaccurate to the final standard).

Reply:
Added the links.